### ISSUE TITLE
AttributeError: 'str' object has no attribute 'get_display_string'

### ISSUE THREAD

**Issue by mloskot** [svenevs/exhale#51]
_(Being Exhale newbie, I'm not sure where the actual problem may be, so I initially posted this as [AttributeError: 'str' object has no attribute 'get_display_string'](https://groups.google.com/forum/#!topic/sphinx-users/1J6HkN-mChE) to sphinx-users list .)_

I'm trying to build Boost.GIL ([sphinx-exhale](https://github.com/boostorg/gil/tree/sphinx-exhale) branch) documentation.  Full logs attached below.

Doxygen and Exhale seems to run fine, as I can see this logged:

```
[+] Exhale: doxygen ran successfully in 11.75 seconds.
[~] Exhale: adding tree view css / javascript.
[+] Exhale: added tree view css / javascript.
[~] Exhale: parsing Doxygen XML.
[+] Exhale: finished parsing Doxygen XML in 35.71 seconds.
[~] Exhale: generating reStructuredText documents.
...
[+] Exhale: generated reStructuredText documents in 23.86 seconds.
```

Then, Sphinx is taking over

```
building [html]: targets for 4515 source files that are out of date
...
```

and it is failing with a very long list of warnings like

```
WARNING: Error when parsing function declaration.
WARNING: Duplicate explicit target name
WARNING: Duplicate ID
WARNING: Duplicate declaration.
WARNING: doxygenfunction: Unable to resolve multiple matches for function ...
```

Eventually failing with

```
Exception occurred:
  File "/mnt/d/boost.wsl/libs/gil/.venv/lib/python3.6/site-packages/sphinx/domains/cpp.py", line 635, in <lambda>
    return self._stringify(lambda ast: ast.get_display_string())
AttributeError: 'str' object has no attribute 'get_display_string'
```

I find it difficult to figure out where is the problem.

@stefanseefeld, my team mate at GIL, posted suggestion to our mailing list, https://lists.boost.org/boost-gil/2018/11/0116.php:

> Just a shot in the dark: duplicate IDs sounds less like a sphinx bug,
> but a bug with the exhale extension, which generates content
> that the sphinx post-processing step then can't consume.

Is this suggestion pinpointing the issue?

I'll greatly appreciate any help.

## Logs

- [sphinx-build-full-output.log](https://github.com/svenevs/exhale/files/2603670/sphinx-build-full-output.log)
- [sphinx-err-29i0rtgw.log](https://github.com/svenevs/exhale/files/2603669/sphinx-err-29i0rtgw.log)

The logs are also available at https://gist.github.com/mloskot/ad6f79b9c5fd15027467e9a55429b52et

**Comment by svenevs** [comment]
Ok cool thanks! I'm testing things locally right now. This project is huge by the way. 

What you said about excluding the `detail` namespace may be necessary (at least at this time. The parser for breathe uses minidom which has memory leaks, we're switching to lxml. This one has taken a lot of time and effort because we also need to fix the testing suite x0

**Comment by mloskot** [comment]
Yes, and currently Doxygen generates docs for everything, including the `detail` namespace. We are going to limit that to the public definitions only, https://lists.boost.org/boost-gil/2018/11/0118.php

I'm on 64 GB RAM and possibly I did not notice memory issues. However, I can see GIL pushes Sphinx and Exhale to the limits :)
